### ISSUE TITLE
Minecart Trains unable to gain momentum if not first in line

### ISSUE THREAD

**Issue by lgnmcrules** [Kipama/Modern_Minecarts#28]
![image](https://github.com/Kipama/Modern_Minecarts/assets/47130259/ea1927b9-63f2-47a5-80c7-6b496aaa7c6b)

Colliding with a part of a chain train with another cart will not transfer momentum and stops the cart.  Even in this case when a powered furnace minecart is attempting to move the train.

1.19.2

**Comment by Kipama** [comment]
So in this case, the furnace minecart is the head of the train, but due to a collision got bumped to the end of the train?
Edit: Ah, I think I got it. The hopper minecart is the head minecart, and the furnace minecart tries to push the train from behind, right?

**Comment by lgnmcrules** [comment]
Yeah, in this case, the hopper minecart is the lead of a 2 cart train consisting of itself and the chest minecart.  A full speed, separate, furnace minecart ran into the chest minecart and stopped without transferring any momentum.

**Comment by Kipama** [comment]
With the way trains are implemented, chained minecarts can only ever be pulled or pushed via the first minecart in a train.
This is unlikely to change in the future, as it would require a full rewrite of the train logic to fix.
